codalab/apps/authenz/models.py

```diff
@@ -32,4 +41,57 @@ class ClUser(auth_models.AbstractUser):
     rabbitmq_username = models.CharField(max_length=36, null=True, blank=True)
     rabbitmq_password = models.CharField(max_length=36, null=True, blank=True)
 
+    def save(self, *args, **kwargs):
+        if self.newsletter_opt_in and self.email and self.is_active:
+            data = {
+                "email_address": self.email,
+                "status": "subscribed",
+            }
+
+            user_hash = hashlib.md5(str.lower(self.email.encode()))
```

Comment:
Will this work? or do you have to explicitly call it with `str`?
```python
user_hash = hashlib.md5(self.email.encode().lower())
```

Reply:
User hash has been added to its own self-contained function in the model